### PR TITLE
Allow only integers for the configuration 'Number of elements to display'

### PR DESCRIPTION
**Pull request by AlexTechDev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Allow only integers for the configuration 'Number of elements to display'
| Type?         | bug fix 
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#28663
| How to test?  | See PrestaShop/Prestashop#28663

### CODE REVIEW

**Comment by okom3pom** [comment]
Hello @AlexTechDev you have to run cs-fixer on your file.

**Comment by MeKeyCool** [comment]
@AlexTechDev you can find a example of command to run : https://github.com/PrestaShop/ps_specials/runs/6897334061?check_suite_focus=true#step:6:1

**Comment by kpodemski** [comment]
@AlexTechDev ping

**Comment by kpodemski** [comment]
@AlexTechDev cs-fixer is still not happy, remember that you can use it locally on your computer

**Comment by AlexTechDev** [comment]
Thanks for the support @kpodemski 
I have some problem with cs-fixer on my computer and I use one validator online but as you say it doesn't work.
I will fix cs-fixer on my computer and upload again

File: ps_specials.php [hunk 1]
```diff
@@ -123,14 +123,22 @@ public function getContent()
...
         if (Tools::isSubmit('submitSpecials')) {
-            Configuration::updateValue('BLOCKSPECIALS_SPECIALS_NBR', (int) Tools::getValue('BLOCKSPECIALS_SPECIALS_NBR'));
+            $nbr = Tools::getValue('BLOCKSPECIALS_SPECIALS_NBR');
+            if (!Validate::isInt($nbr) || $nbr <= 0) {
+                $errors = $this->trans('The number of products is invalid. Please enter a positive number.', [], 'Modules.Specials.Admin');
+            }
+            if (!empty($errors)) {
+                $output = $this->displayError($errors);
+            } else {
+                Configuration::updateValue('BLOCKSPECIALS_SPECIALS_NBR', (int)Tools::getValue('BLOCKSPECIALS_SPECIALS_NBR'));
```
Comment (SharakPL) [inline ps_specials.php:133]:
```suggestion
                Configuration::updateValue('BLOCKSPECIALS_SPECIALS_NBR', (int) Tools::getValue('BLOCKSPECIALS_SPECIALS_NBR'));
```

File: ps_specials.php [hunk 2]
```diff
@@ -123,14 +123,22 @@ public function getContent()
...
 
-            $this->_clearCache('*');
+                $this->_clearCache('*');
 
-            $output .= $this->displayConfirmation($this->trans('The settings have been updated.', [], 'Admin.Notifications.Success'));
+                $output .= $this->displayConfirmation($this->trans('The settings have been updated.', array(), 'Admin.Notifications.Success'));
```
Comment (SharakPL) [inline ps_specials.php:137]:
```suggestion
                $output .= $this->displayConfirmation($this->trans('The settings have been updated.', [], 'Admin.Notifications.Success'));
```

File: ps_specials.php [hunk 3]
```diff
@@ -123,14 +123,22 @@ public function getContent()
...
+            }
         }
 
-        return $output . $this->renderForm();
+        return $output.$this->renderForm();
```
Comment (SharakPL) [inline ps_specials.php:141]:
```suggestion
        return $output . $this->renderForm();
```

**Review by SharakPL — CHANGES_REQUESTED** [review]
These should fix CS

**Comment by Hlavtox** [comment]
@AlexTechDev Fixed the CS, hope you don't mind. 🙏

**Review by SharakPL — APPROVED** [review]

**Review by djoelleuch — APPROVED** [review]
Hello @AlexTechDev ,

Tested with the following PS versions and PHP 7.4 :
 - PS 8.0.1 :heavy_check_mark: 
 - PS 1.7.8.8 :heavy_check_mark: 
 - PS 8.0.0 :heavy_check_mark: 
 
PR2

It's QA approved  ! 

Thank you for your PR :rocket:

**Comment by Hlavtox** [comment]
Thanks @AlexTechDev & @djoelleuch! :-)

**Pull request merged**